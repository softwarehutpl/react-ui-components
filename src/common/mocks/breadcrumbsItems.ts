export default [{
  name: 'Home', link: 'link',
  }, {
    name: 'Products', link: 'link-2',
  }, {
    name: 'Collection', link: 'link-3',
  }, {
    name: 'Product Details', link: 'link-4',
  }];
