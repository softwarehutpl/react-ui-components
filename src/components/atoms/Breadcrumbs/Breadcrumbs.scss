@import '../../../common/styles/variables.scss';

// TODO: remove this when remove example
.active {
  font-weight: bold;
  font-family: 'Amiri', serif;
}

.breadcrumbsWrapper {
  padding: 0 $metric-s;
}

.breadcrumbsItem {
  font-family: 'Amiri', serif;
  &:hover {
    color: $color-piper;
  }
}
  