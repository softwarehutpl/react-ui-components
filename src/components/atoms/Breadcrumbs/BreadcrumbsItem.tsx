import React from 'react';
import styled from 'styled-components';

interface BreadcrumbsItemContainerProps {
  // onClick: (e: any) => void;
  link: string;
  name: string;
  items: any;
  isActive: boolean;
};

interface BreadcrumbsItemButtonProps {
  link: string;
  name: string;
  items: any;
};

interface BreadcrumbsProps extends BreadcrumbsItemContainerProps {
  items: any;
  link: string;
  name: string;
};

export const BreadcrumbsItemContainer = styled.div<BreadcrumbsItemContainerProps>`
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 10px;
`;

export const BreadcrumbsItemButton = styled.button<BreadcrumbsItemButtonProps>`
  margin-bottom: 10px;
  font-weight: bold;
  outline: none;
  border: none;
  background-color: transparent;
`;

const BreadcrumbsItem = (props: BreadcrumbsProps) => {
  const goToLink = (e: any) => {
    console.log(e);
    window.history.pushState(null, 'link', props.link);
  }
  return (
    <BreadcrumbsItemContainer {...props}>
      <BreadcrumbsItemButton {...props} onClick={goToLink}>
        {props.name}
      </BreadcrumbsItemButton>
    </BreadcrumbsItemContainer>
  )};

const defaultProps = {
  items: null,
};

BreadcrumbsItem.defaultProps = defaultProps;

export default BreadcrumbsItem;
