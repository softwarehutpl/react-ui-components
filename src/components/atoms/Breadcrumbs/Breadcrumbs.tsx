import React from 'react';
import styled from 'styled-components';
import BreadcrumbsItem from './BreadcrumbsItem';

interface BreadcrumbsContainerProps {
};

interface BreadcrumbsProps extends BreadcrumbsContainerProps {
  items: any;
  showOnlyBorderItems: boolean;
};

export const BreadcrumbsContainer = styled.div<BreadcrumbsContainerProps>`
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
`;

const Breadcrumbs = (props: BreadcrumbsProps) => {
  return (
    <BreadcrumbsContainer>
      {props.items && props.items.length > 0 && props.items.map((item: any, index: number) => {
        if (!props.showOnlyBorderItems || (props.showOnlyBorderItems &&
          (index === 0 || index === props.items.length - 1))) {
          return (<BreadcrumbsItem
            key={item.link}
            isActive={index === props.items.length - 1}
            name={item.name}
            link={item.link}
            items={item.items} />)
        }
})}
    </BreadcrumbsContainer>
  )};

const defaultProps = {
  color: 'primary',
  fontColor: '#000',
  showOnlyBorderItems: false,
  disabled: false,
  className: '',
  borderWidth: 1,
  margin: 0,
  padding: 15,
  borderRadius: 5,
  showPlaceholderOnFocus: false,
  width: 200,
  direction: 'column',
  labelPosition: 'left',
  backgroundColor: '#FFF',
};

Breadcrumbs.defaultProps = defaultProps;

export default Breadcrumbs;
