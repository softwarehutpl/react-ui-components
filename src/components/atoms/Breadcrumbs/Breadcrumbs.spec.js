
import React from 'react';
import { create, act } from 'react-test-renderer';
import 'jest-styled-components';
import breadcrumbsItems from '../../../common/mocks/breadcrumbsItems';
import Breadcrumbs, { BreadcrumbsContainer, BreadcrumbsGap } from './Breadcrumbs';
import { BreadcrumbsItemContainer } from './BreadcrumbsItem';

describe('Breadcrumbs Component', () => {
  it('should render Breadcrumbs correctly', () => {
    const tree = create(<Breadcrumbs items={breadcrumbsItems} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render BreadcrumbsItemContainer correctly', () => {
    const tree = create(<BreadcrumbsItemContainer items={breadcrumbsItems} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call button onClick method in BreadcrumbsGap', () => {
    const mockOnChange = jest.fn();
    const component = create(<BreadcrumbsGap onClick={mockOnChange} />);
    const instance = component.root;
    const button = instance.findByType('button');
    act(() => {
      button.props.onClick();
    });
    expect(mockOnChange.mock.calls.length).toEqual(1);
  });

  it('should render BreadcrumbsContainer with introduced params', () => {
    const component = create(<BreadcrumbsContainer
      borderWidth={2}
      backgroundColor='hotpink'
      borderRadius={3}
      margin={20}
    />);
    const breadcrumbsContainer = component.toJSON();
    expect(breadcrumbsContainer).toHaveStyleRule('background-color', 'hotpink');
    expect(breadcrumbsContainer).toHaveStyleRule('border-width', '2px');
    expect(breadcrumbsContainer).toHaveStyleRule('border-radius', '3px');
    expect(breadcrumbsContainer).toHaveStyleRule('margin', '20px');
  });
  
  it('should render BreadcrumbsContainer without border', () => {
    const component = create(<BreadcrumbsContainer noBorder />);
    const breadcrumbsContainer = component.toJSON();
    expect(breadcrumbsContainer).toHaveStyleRule('border', 'none');
  });

  it('should render BreadcrumbsContainer with default padding and margin', () => {
    const component = create(<BreadcrumbsContainer />);
    const breadcrumbsContainer = component.toJSON();
    expect(breadcrumbsContainer).toHaveStyleRule('padding', '0 15px');
    expect(breadcrumbsContainer).toHaveStyleRule('margin', '0px');
  });
});
