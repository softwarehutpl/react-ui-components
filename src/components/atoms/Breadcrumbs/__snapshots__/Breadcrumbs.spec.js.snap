// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs Component should render Breadcrumbs correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
  border: none;
  background-color: transparent;
  padding: 15px 10px;
  cursor: pointer;
  margin: 0;
  font-size: 16px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 50%;
  -ms-flex: 0 0 50%;
  flex: 0 0 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: auto;
  jusity-content: flex-start;
  border: solid;
  border-width: 1px;
  background-color: #fff;
  padding: 0 15px;
  margin: 0px;
  border-color: #8e8e8e;
  border-radius: 0px;
  position: relative;
}

<div
  className="c0"
  color="#8e8e8e"
  disabled={false}
>
  <div
    className="c1"
    color="#8a8a8a"
    fontSize={16}
    name="Home"
  >
    <button
      className="c2 undefined undefined"
      color="#8a8a8a"
      fontSize={16}
      name="Home"
      onClick={[Function]}
    >
      Home
    </button>
    <span
      className="c3"
    >
      /
    </span>
  </div>
  <div
    className="c1"
    color="#8a8a8a"
    fontSize={16}
    name="Products"
  >
    <button
      className="c2 undefined undefined"
      color="#8a8a8a"
      fontSize={16}
      name="Products"
      onClick={[Function]}
    >
      Products
    </button>
    <span
      className="c3"
    >
      /
    </span>
  </div>
  <div
    className="c1"
    color="#8a8a8a"
    fontSize={16}
    name="Collection"
  >
    <button
      className="c2 undefined undefined"
      color="#8a8a8a"
      fontSize={16}
      name="Collection"
      onClick={[Function]}
    >
      Collection
    </button>
    <span
      className="c3"
    >
      /
    </span>
  </div>
  <div
    className="c1"
    color="#8a8a8a"
    fontSize={16}
    name="Product Details"
  >
    <button
      className="c2 undefined undefined"
      color="#8a8a8a"
      fontSize={16}
      name="Product Details"
      onClick={[Function]}
    >
      Product Details
    </button>
  </div>
</div>
`;

exports[`Breadcrumbs Component should render BreadcrumbsItemContainer correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;
